src/app/auth/auth-registration/auth-registration.component.ts

```diff
@@ -0,0 +1,278 @@
+import { Component, OnInit } from "@angular/core";
+import { Router, ActivatedRoute, ParamMap } from "@angular/router";
+import { UtilsService } from "@services/utils.service";
+import { NotificationService } from '@shared/notification/notification.service';
+import { Md5 } from "ts-md5/dist/md5";
+import {
+  Validators,
+  FormControl,
+  FormGroup,
+  FormBuilder
+} from "@angular/forms";
+
+import { AuthService } from "../auth.service";
+import { BrowserStorageService } from "@services/storage.service";
+
+@Component({
+  selector: "app-auth-registration",
+  templateUrl: "./auth-registration.component.html",
+  styleUrls: ["./auth-registration.component.scss"]
+})
+export class AuthRegistrationComponent implements OnInit {
+  password: string;
+  confirmPassword: string;
+  isAgreed: boolean = false;
+  registerationForm: FormGroup;
+  hide_password: boolean = false;
+  user: any = {
+    email: null,
+    activation_code: null,
+    contact: null
+  };
+  domain = window.location.hostname;
+  // validation errors array
+  errors: Array<any> = [];
+
+  constructor(
+    private route: ActivatedRoute,
+    private authService: AuthService,
+    private utils: UtilsService,
+    private storage: BrowserStorageService,
+    private notificationService: NotificationService
+  ) {
+    this.initForm();
+  }
+
+  ngOnInit() {
+    this.domain =
+      this.domain.indexOf("127.0.0.1") !== -1 ||
+      this.domain.indexOf("localhost") !== -1
+        ? "appdev.practera.com"
+        : this.domain;
+    this.validateQueryParams();
+  }
+
+  initForm() {
+    this.registerationForm = new FormGroup({
+      password: new FormControl("", [
+        Validators.required,
+        Validators.minLength(8)
+      ]),
+      confirmPassword: new FormControl("", [Validators.required])
+    });
+  }
+
+  validateQueryParams() {
+    // access query params
+    this.route.queryParamMap.subscribe(queryParams => {
+      let email = queryParams.get("email");
+      this.user.email = email;
+      let activation_code = queryParams.get("activation_code");
+      if (email && activation_code) {
+        // set query params to stroage
+        this.storage.set("hash", {
+          email: email,
+          key: activation_code
+        });
+        // check is Url valid or not.
+        this.authService
+          .verifyRegistration({
+            email: this.storage.get("hash").email,
+            key: this.storage.get("hash").key
+          }).subscribe(response => {
+              if (response) {
```

Comment:
reason for this in request service error handler return the value as it’s a success value. So component error call backs are not calling because of that. I had a chat with caw and i have create a story for this issue. [`AV2-122`](https://intersective.atlassian.net/browse/AV2-122)